scripts/create_dataset.py

```diff
@@ -1,69 +1,160 @@
+                                                                  input_tables=input_tables)
+        tables.append(table)
+        relationships.extend(new_relationships)
+
+    return tables, relationships
+
+
+def create_table_and_relationships(input_table, input_tables):
+    """
+    Creates a table and relationships using the input tables fields and reference field.
+    :param input_table: the input table to be used to create a new table and relationships
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+
+    input_table_name = input_table["name"]
+    columns, table_relationships = [], []
+    for input_column in input_table["columns"]:
+        column, new_relationships = create_column_and_relationships(input_relationships=input_column.get("links"),
+                                                                    column_name=input_column["name"],
+                                                                    column_datatype=input_column["datatype"],
+                                                                    column_array_of=input_column.get("array_of"),
+                                                                    input_table_name=input_table_name,
+                                                                    input_tables=input_tables)
+        columns.append(column)
+        table_relationships.extend(new_relationships)
+
+    table = {"name": input_table_name, "columns": columns}
+
+    return table, table_relationships
+
+
+def create_column_and_relationships(input_relationships,
+                                    column_name,
+                                    column_datatype,
+                                    column_array_of,
+                                    input_table_name,
+                                    input_tables):
+    """
+    Creates a column and relationships, keeping ony the name, data type and column field from the input column.
+    :param input_relationships: the list of relationships from an input JSON column
+    :param column_name: the name of the input JSON column
+    :param column_datatype: the primitive type enforced for values of the input JSON column
+    :param column_array_of: indicates if the column values should be in array
+    :param input_table_name: the name of input JSON table that has the input_column
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    if not any([validate_table_and_column(table_name=input_relationship["table_name"],
+                                          column_name=input_relationship["column_name"],
+                                          column_datatype=column_datatype,
+                                          input_tables=input_tables)
+                for input_relationship in input_relationships] if input_relationships else [True]):
```

Comment:
I think this could be a little more readable, by replacing the `if input_relationships else [True]` with a pre-check in the enclosing `if`:
```python
if input_relationships and not any([... stuff ...])
```